CONTRIBUTING.md

```diff
@@ -137,5 +137,42 @@ TODO: review/complete; see #31
 
 ## <a name="commit"></a> Commit Message Guidelines
 We have precise rules over how our git commit messages can be formatted. This leads to **more readable messages** that are easy to follow when looking through the **project history**.
+We also use the git commit messages to generate our changelog.
 
-TODO document commit message guidelines: #28
+We're using Angular's commit message format: `type(scope): subject`
+
+#### Type
+Must be one of the following:
+
+* **feat**: A new feature
+* **fix**: A bug fix
+* **docs**: Documentation only changes
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **perf**: A code change that improves performance
+* **test**: Adding missing tests
+* **chore**: Changes to the build process or auxiliary tools and libraries such as documentation generation
+
+#### Scope
+The scope can be anything specifying place of the commit change. For example `stark-core`, `table`, `theme`, `sidenav`, etc. If you make multiple commits for the same component, please keep the naming of this component consistent. For example, if you make a change to navigation and the first commit is `fix(sidenav)`, you should continue to use `sidenav` for any more commits related to navigation.
+
+#### Subject
+The subject contains succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* do not capitalize first letter
+* do not place a period `.` at the end
+* entire length of the commit message must not go over 50 characters
+* describe what the commit does, not what issue it relates to or fixes
+* **be brief, yet descriptive** - we should have a good understanding of what the commit does by reading the subject
+
+### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+### Footer
+The footer should contain any information about **Breaking Changes** and is also the place to
+[reference GitHub issues that this commit closes](https://help.github.com/articles/closing-issues-via-commit-messages/).
+
+**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines.
```

Comment:
It is not clear to me where exactly should be put this "breaking change", in the footer? body?

Reply:
I've added some examples taken from other projects

Reply:
Wow, now we do have examples ;)